### PR TITLE
[search] Support suburbs in geocoder.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTALL

**Comment by tatiana-yan** [comment]
JTALL

**Comment by tatiana-yan** [comment]
JTALL

File: search/features_layer_matcher.hpp [hunk 1]
```diff
@@ -365,6 +370,18 @@ class FeaturesLayerMatcher
     }
   }
 
+  template <typename Fn>
+  void MatchStreetsWithSuburbs(FeaturesLayer const & child, FeaturesLayer const & parent, Fn && fn)
+  {
+    // We have pre-matched streets from geocoder.
+    auto const & streets = *child.m_sortedFeatures;
+    CHECK_EQUAL(parent.m_sortedFeatures->size(), 1, ());
+    auto const & suburb = (*parent.m_sortedFeatures)[0];
```
Comment (mpimenov) [inline search/features_layer_matcher.hpp]: nit: front()?

File: search/geocoder.cpp [hunk 2]
```diff
@@ -1067,18 +1074,54 @@ void Geocoder::GreedilyMatchStreets(BaseContext & ctx)
...
 
+  // Match streets without suburbs.
   vector<StreetsMatcher::Prediction> predictions;
-  StreetsMatcher::Go(ctx, *m_filter, m_params, predictions);
+  StreetsMatcher::Go(ctx, ctx.m_streets, *m_filter, m_params, predictions);
 
   for (auto const & prediction : predictions)
     CreateStreetsLayerAndMatchLowerLayers(ctx, prediction);
+
+  // Match streets suburbs.
```
Comment (mpimenov) [inline search/geocoder.cpp]: мб различать ветки в Trace?

Comment (tatiana-yan) [reply search/geocoder.cpp]: done

**Comment by tatiana-yan** [comment]
JTLINUX

File: search/features_layer_matcher.hpp (hunk 1 above)
Comment (tatiana-yan) [reply search/features_layer_matcher.hpp]: done

File: indexer/ftypes_matcher.hpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ class IsAirportChecker : public BaseChecker
 class IsSquareChecker : public BaseChecker
 {
   IsSquareChecker();
-  friend class IsStreetOrSuburbChecker;
+  friend class IsStreetOrSquareChecker;
```
[the remark this answers is not in the document]
Comment (tatiana-yan) [reply indexer/ftypes_matcher.hpp]: done

**Comment by tatiana-yan** [comment]
JTLINUX

**Comment by tatiana-yan** [comment]
JTLINUX

**Pull request merged**